### DISCUSSION TITLE
Many to many inner_join query inferred types

### DISCUSSION THREAD

**Discussion by onx2** [diesel-rs/diesel#3367] [Q&A] [answered]
**Update (10-09-2022) - modified question**

How can I get the results of an `inner_join` query to be typed correctly?

Reproduction repo:
https://github.com/onx2/diesel-n-m

The results [here](https://github.com/onx2/diesel-n-m/blob/main/src/main.rs#L34) are `Result<{unknown}, {unknown}>` even though I'm passing a type def into `load`.

**Comment by weiznich** [comment]
I'm sorry, but its impossible to answer such questions without having access to all necessary information. In the best case you are able to provide a reproducible example (something someone can run `cargo check` in and see the problem). Otherwise at least the following information are required:
* diesel and rustc versions (including feature flags for diesel)
* versions of any involved third party crate
* all relevant code, which includes the definition/import of any type thats not from diesel
* the relevant part of your schema.rs file (all involved tables + joinable + ...)
* the code that generates the error message
* the exact error message as emitted by rustc

**Comment by onx2** [comment]
Got it - I realized the question wasn't even what I wanted to ask to start so I modified it and added a reproduction repo. Once I understand some more of this I think I'll need to branch off into new discussions but I'd like to start simple 😄 

Thanks in advance for any help you can give and thanks for the awesome ORM.

**Reply by onx2** [reply]
Or is this simply something that must be explicitly labeled with diesel?

let result: QueryResult<Vec<(FormQuestion, Question)>> = ...the inner_join  query;

**Comment by weiznich** [comment, marked as answer]
> The results here are Result<{unknown}, {unknown}> even though I'm passing a type def into load.

That sounds like you refer to type hints shown by rust-analyzer. In that case: If it compiles fine with rustc and rust-analyzer is not able to show the correct type, thats a bug in rust-analyzer and should be reported in their issue tracker.

**Reply by onx2** [reply]
Got it - thank you
